### ISSUE TITLE
Rule Change: Marking constructors private is not "useless"

### ISSUE THREAD

**Issue by TiesWestendorp** [eslint/eslint#16825]
### What rule do you want to change?

no-useless-constructor

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
private constructor(arg) {
  super(arg);
}
```

### What does the rule currently do for this code?

It generates a warning, based on the fact that the only call inside the constructor is `super`.

### What will the rule do after it's changed?

No longer generate an error; changing the visibility of the constructor is a significant change. "Uselessness" is not a property that can be determined solely from the constructor body.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @TiesWestendorp, thanks for the issue!

The core `no-useless-constructor` rule does not support typescript-specific syntax. For typescript code, instead of the core rule you should use the typescript-eslint plugin rule `@typescript-eslint/no-useless-constructor`.

https://typescript-eslint.io/rules/no-useless-constructor

Per the documentation for this rule, it adds support for constructors marked as `protected`/`private`.

**Comment by TiesWestendorp** [comment]
My bad, wasn't aware of this. Thanks for explaining it!
